src/platform/platform.ts

```diff
@@ -581,30 +581,30 @@ export class Platform {
         if (win['innerWidth'] < win['innerHeight']) {
 
           // the device is in portrait
+          // we have to do fancier checking here
+          // because of the virtual keyboard resizing
+          // the window
           if (this._pW <= win['innerWidth']) {
             console.debug('setting _isPortrait to true');
             this._isPortrait = true;
             this._pW = win['innerWidth'];
           }
+
           if (this._pH <= win['innerHeight']) {
             console.debug('setting _isPortrait to true');
             this._isPortrait = true;
             this._pH = win['innerHeight'];
           }
 
         } else {
-          if (this._lW > win['innerWidth']) {
-            // Special case: keyboard is open and device is in portrait
-            console.debug('setting _isPortrait to true while keyboard is open and device is portrait');
-            this._isPortrait = true;
-          }
           // the device is in landscape
-          if (this._lW <= win['innerWidth']) {
+          if (this._lW !== win['innerWidth']) {
```

Comment:
could `win['innerWidth']` be saved in a local variable, so it does not have to be called twice?

same with `win['innerHeight']`